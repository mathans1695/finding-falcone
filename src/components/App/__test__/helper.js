export const mockHandleResult = (selectedPlanets, selectedVehicles) => undefined;

export const selectedPlanets = ['Donlon', 'Enchai', 'Sapir', 'Pingasor'];
export const selectedVehicles = ['Space pod', 'Space rocket', 'Space ship', 'Space ship'];