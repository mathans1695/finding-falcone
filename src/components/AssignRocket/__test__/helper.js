import _ from 'lodash';

const vehicles = [
	{name: "Space pod", total_no: 2, max_distance: 200, speed: 2, isPossible: false, showAlways: false},
	{name: "Space rocket", total_no: 1, max_distance: 300, speed: 4, isPossible: false, showAlways: false },
	{name: "Space shuttle", total_no: 1, max_distance: 400, speed: 5, isPossible: false, showAlways: false },
	{name: "Space ship", total_no: 2, max_distance: 600, speed: 10, isPossible: false, showAlways: false }
];
const id = '111111';

const generateVehicles = (planetDistance, selected) => {
	let vehiclesObj = {
		id: id,
		planetDistance: planetDistance,
		vehicles: _.cloneDeep(vehicles),
		previousSelected: ''
	}
	
	vehiclesObj.vehicles.forEach(vehicle => {
		if(vehicle.max_distance >= planetDistance && vehicle.total_no > 0) {
			vehicle.isPossible = true;
		}
	});
	
	if(selected) {
		vehiclesObj.vehicles.forEach(vehicle => {
			if(vehicle.isPossible && vehicle.name === selected){
				vehicle.showAlways = true;
				vehicle.total_no -= 1;
				vehiclesObj.previousSelected = selected;
			}
		})
	}
	
	return vehiclesObj
}

export const initialVehicles = generateVehicles(200);
export const selectVehicles = generateVehicles(200, 'Space pod');
export const disableParticularVehicle = generateVehicles(400, 'Space pod');

export const event = {
	target: {
		value: 'Space rocket',
		getAttribute: (attr) => {
			if(attr === 'data-id') return '111111';
			if(attr === 'data-speed') return 4;
		}
	}
}

export const mockHandleVehicleUpdation = jest.fn((id, rocket, speed, distance) => undefined);