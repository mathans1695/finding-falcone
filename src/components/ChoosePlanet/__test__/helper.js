const planets = [
	{name: "Donlon", distance: 100},
	{name: "Enchai", distance: 200},
	{name: "Jebing", distance: 300},
	{name: "Sapir", distance: 400},
	{name: "Lerbin", distance: 500},
	{name: "Pingasor", distance: 600}
]
const id = "111111";

export const initialPlanets = {
	curPlanet: "Choose Planet",
	id: id,
	planets: planets,
	previousSelected: []
}

export const changePlanets = {
	curPlanet: "Donlon",
	id: id,
	planets: planets,
	previousSelected: [{name: "Donlon", distance: 100}]
}

export const mockUpdateListOfPlanets = jest.fn((id, value, distance) => undefined);
export const mockUpdateListOfVehicles = jest.fn((id, distance) => undefined);

export const event = {
	target: {
		parentElement: { id },
		options: [
			{ selected: false },
			{ selected: false },
			{ selected: true, getAttribute: () => 200 },
			{ selected: false },
			{ selected: false },
			{ selected: false },
			{ selected: false }
		],
		value: 'Enchai'
	}
}