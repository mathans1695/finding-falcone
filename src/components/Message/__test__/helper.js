export const resetMsg = 'Reset successful';
export const notAvailableMsg = 'Selected rocket is not available';