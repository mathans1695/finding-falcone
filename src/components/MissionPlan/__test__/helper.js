export const globalVehicles = [
	{name: "Space pod", total_no: 2, max_distance: 200, speed: 2 },
	{name: "Space rocket", total_no: 1, max_distance: 300, speed: 4 },
	{name: "Space shuttle", total_no: 1, max_distance: 400, speed: 5 },
	{name: "Space ship", total_no: 2, max_distance: 600, speed: 10 }
];

export const indivVehicles = [
	{name: "Space pod", total_no: 2, max_distance: 200, speed: 2, isPossible: false, showAlways: false},
	{name: "Space rocket", total_no: 1, max_distance: 300, speed: 4, isPossible: false, showAlways: false },
	{name: "Space shuttle", total_no: 1, max_distance: 400, speed: 5, isPossible: false, showAlways: false },
	{name: "Space ship", total_no: 2, max_distance: 600, speed: 10, isPossible: false, showAlways: false }
];

export const getIndex = (vehicles, name) => vehicles.findIndex(vehicle => vehicle.name === name);