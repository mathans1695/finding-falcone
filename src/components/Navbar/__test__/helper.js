export const mockReset = jest.fn();