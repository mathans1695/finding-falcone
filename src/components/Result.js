import React, { Component } from 'react';
import loading from '../Images/loading.gif';
import { Link } from 'react-router-dom';
import '../styles/Result.css';
import Navbar from './Navbar';
import Footer from './Footer';

class Result extends Component {
	render() {
		const {resultJSON, time, reset} = this.props;
		console.log(resultJSON);
		
		return (
			<div className='Result'>
				{resultJSON ?
				  resultJSON.status === 'success' ?
					  <>
						<p className='Result__msg'>Success! Congratulations on Finding Falcone. King Shan is mighty pleased.</p>
						<p className='Result__time'>
							Time taken: {time.reduce((a, b) => {
								return a+b;
							}, 0)}
						</p>	
						<p className='Result__planet'>
							Planet found: {resultJSON.planet_name}
						</p>
						<Link to='/'>
							<button 
								className='Result__startAgain'
								onClick={reset}
							>
								Start Again
							</button>
						</Link>
					  </>
					: <>
						<p className='Result__msg'>Mission Failed!. King Shan is mad at you.</p>
						<p className='Result__time'>
							Time taken: {time.reduce((a, b) => {
								return a+b;
							}, 0)}
						</p>
						<Link to='/'>
							<button 
								className='Result__startAgain'
								onClick={reset}
							>
								Start Again
							</button>
						</Link>
					  </>
				  : <img src={loading} alt='laoding' />
				}
			</div>
		)
	}
}

export default Result;