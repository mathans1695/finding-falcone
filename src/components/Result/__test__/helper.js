export const successJson = {
	planet_name: "Donlon",
	status: "success"
}

export const failureJson = {
	status: "false"
}

export const time = [50, 30, 50, 80];

export const total = 210;

export const mockReset = jest.fn();