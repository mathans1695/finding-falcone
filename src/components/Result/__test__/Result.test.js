import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Result from '../Result';
import { successJson, failureJson, time, total, mockReset} from './helper.js';

it('Renders success result page correctly', () => {
	const wrapper = shallow(<Result resultJSON={successJson} time={time} />);
	
	expect(toJson(wrapper)).toMatchSnapshot();
})

it('Renders failure result page correctly', () => {
	const wrapper = shallow(<Result resultJSON={failureJson} time={time} />);
	
	expect(toJson(wrapper)).toMatchSnapshot();
})

it('Time taken equals the sum of time props', () => {
	const wrapper = shallow(<Result resultJSON={failureJson} time={time} />);
	const timeTakenText = wrapper.find('.Result__time').text();
	
	expect(timeTakenText).toEqual(`Time taken: ${total}`);
})

it('Planet found equals the planet_names property of resultJSON', () => {
	const wrapper = shallow(<Result resultJSON={successJson} time={time} />);
	const planetFoundText = wrapper.find('.Result__planet').text();
	
	expect(planetFoundText).toEqual(`Planet found: ${successJson.planet_name}`);
})

it('Start Again should call reset function passed as prop', () => {
	const wrapper = shallow(<Result resultJSON={successJson} time={time} reset={mockReset} />);
	wrapper.find('.Result__startAgain').simulate('click');
	
	expect(mockReset.mock.calls.length).toEqual(1);
})