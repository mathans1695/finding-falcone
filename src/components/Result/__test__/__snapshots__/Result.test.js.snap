// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders failure result page correctly 1`] = `
<div
  className="Result"
>
  <p
    className="Result__msg"
  >
    Mission Failed!. King Shan is mad at you.
  </p>
  <p
    className="Result__time"
  >
    Time taken: 
    210
  </p>
  <Link
    to="/"
  >
    <button
      className="Result__startAgain"
    >
      Start Again
    </button>
  </Link>
</div>
`;

exports[`Renders success result page correctly 1`] = `
<div
  className="Result"
>
  <p
    className="Result__msg"
  >
    Success! Congratulations on Finding Falcone. King Shan is mighty pleased.
  </p>
  <p
    className="Result__time"
  >
    Time taken: 
    210
  </p>
  <p
    className="Result__planet"
  >
    Planet found: 
    Donlon
  </p>
  <Link
    to="/"
  >
    <button
      className="Result__startAgain"
    >
      Start Again
    </button>
  </Link>
</div>
`;
